node/lib/util/submodule_config_util.js

```diff
@@ -71,6 +71,34 @@ const CONFIG_FILE_NAME = "config";
 exports.modulesFileName = ".gitmodules";
 
 /**
+ * Return the relative path from the working directory of a submodule having
+ * the specified `subName` to its .git directory.
+ * @param {String} subName
+ */
+exports.computeRelativeGitDir = function (subName) {
```

Comment:
It would change over time because the .gitmodules file changes.

But re config file, in that case, I still have no idea what the config file is for, or how the correct mapping from dir to module is supposed to work.  I guess this will do for now, since we're not doing anything truly destructive.